File: 08-greedy_algorithms/to_implement/greedy_sets-TO_IMPLEMENT.rb
require "rspec/autorun"
require "set"

def greedy_search(states_needed, stations)
    # TODO:
end

# tests here
describe 'greedy search' do
    it "tests that the greedy search works" do
        states_needed = Set.new(['mt', 'wa', 'or', 'id', 'nv', 'ut', 'ca', 'az'])

        stations = {
            'kone' => Set.new(['id', 'nv', 'ut']),
            'ktwo' => Set.new(['wa', 'id', 'mt']),
            'kthree' => Set.new(['or', 'nv', 'ca']),
            'kfour' => Set.new(['nv', 'ut']),
            'kfive' => Set.new(['ca', 'az']),
        }
        result = greedy_search(states_needed, stations)
        expect(result).to eq(['kone', 'ktwo', 'kthree', 'kfive'])
    end

    # TODO: add more test cases
end

File: 08-greedy_algorithms/to_implement/greedy_sets-TO_IMPLEMENT.js
const testEquality = require('../../testingFunction');

const getIntersection = (set1, set2) => {
    return new Set([...set1].filter(item => set2.has(item)));
}

const getDifference = (set1, set2) => {
    return new Set([...set1].filter(item => !set2.has(item)));
}

const getUnion = (set1, set2) => {
    return new Set([...set1].concat([...set2]));
}

statesNeeded = new Set(['mt', 'wa', 'or', 'id', 'nv', 'ut', 'ca', 'az'])

stations = {
    kone: new Set(['id', 'nv', 'ut']),
    ktwo: new Set(['wa', 'id', 'mt']),
    kthree: new Set(['or', 'nv', 'ca']),
    kfour: new Set(['nv', 'ut']),
    kfive: new Set(['ca', 'az']),
}

// TODO:

testEquality(finalStations, ['kone', 'ktwo', 'kthree', 'kfive']);
